src/main/java/com/example/service/BookServiceImpl.java

```diff
@@ -34,4 +34,18 @@ public BookDto getBookById(Long id) {
                 () -> new EntityNotFoundException("Can't not find book by id " + id));
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateBookById(Long id, CreateBookRequestDto requestDto) {
+        bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't not find book by id " + id));
+        Book book = bookMapper.toModel(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
```

Comment:
Let’s  add the mapper method 
`Book updateModel(CreateBookRequestDto requestDto, Book book)`